qiskit_experiments/framework/restless_mixin.py

```diff
@@ -128,20 +138,49 @@ def enable_restless(
                 f"a smaller repetition delay for the restless experiment."
             )
 
-    def _get_restless_processor(self) -> DataProcessor:
+    def _get_restless_processor(self, meas_level: int = 2) -> DataProcessor:
         """Returns the restless experiments data processor.
 
         Notes:
             Sub-classes can override this method if they need more complex data processing.
         """
         outcome = self.analysis.options.get("outcome", "1" * self._num_qubits)
-        return DataProcessor(
-            "memory",
-            [
-                nodes.RestlessToCounts(self._num_qubits),
-                nodes.Probability(outcome),
-            ],
+        normalize = self.analysis.options.get("normalization", True)
+        dimensionality_reduction = self.analysis.options.get(
+            "dimensionality_reduction", ProjectorType.SVD
         )
+        # Todo: The following code is copied and adapted from the processor library.
+        # Todo: This could maybe be simplified.
+        if meas_level == MeasLevel.KERNELED:
+
+            try:
+                if isinstance(dimensionality_reduction, ProjectorType):
+                    projector_name = dimensionality_reduction.name
+                else:
+                    projector_name = dimensionality_reduction
+                projector = ProjectorType[projector_name].value
+
+            except KeyError as error:
+                raise DataProcessorError(
+                    f"Invalid dimensionality reduction: {dimensionality_reduction}."
+                ) from error
```

Comment:
If this is copied from somewhere then that could be justification enough to outsource this to a utility function.

Reply:
I outsourced the code here: 4654d79